assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py

```diff
@@ -218,30 +219,42 @@ def get_model_signature(self) -> ModelSignature:
                         CLIPMLFlowSchemaLiterals.INPUT_COLUMN_TEXT),
             ]
         )
-        output_schema = Schema(
-            [
-                ColSpec(CLIPMLFlowSchemaLiterals.OUTPUT_COLUMN_DATA_TYPE,
-                        CLIPMLFlowSchemaLiterals.OUTPUT_COLUMN_PROBS),
-                ColSpec(CLIPMLFlowSchemaLiterals.OUTPUT_COLUMN_DATA_TYPE,
-                        CLIPMLFlowSchemaLiterals.OUTPUT_COLUMN_LABELS),
-            ]
-        )
+
+        output_schema = None
+        if self._task == SupportedTasks.ZERO_SHOT_IMAGE_CLASSIFICATION.value:
+            output_schema = Schema(
+                [
+                    ColSpec(CLIPMLFlowSchemaLiterals.OUTPUT_COLUMN_DATA_TYPE,
+                            CLIPMLFlowSchemaLiterals.OUTPUT_COLUMN_PROBS),
+                    ColSpec(CLIPMLFlowSchemaLiterals.OUTPUT_COLUMN_DATA_TYPE,
+                            CLIPMLFlowSchemaLiterals.OUTPUT_COLUMN_LABELS),
+                ]
+            )
 
         return ModelSignature(inputs=input_schema, outputs=output_schema)
 
     def save_as_mlflow(self):
         """Prepare model for save to MLflow."""
         sys.path.append(self.MODEL_DIR)
-        from mlflow_wrapper import CLIPMLFlowModelWrapper
-
+        from clip_mlflow_wrapper import CLIPMLFlowModelWrapper
         mlflow_model_wrapper = CLIPMLFlowModelWrapper(task_type=self._task)
+        mlflow_model_wrapper_path = os.path.join(self.MODEL_DIR, "clip_mlflow_wrapper.py")
```

Comment:
Minor: Can add a function _get_code_paths and have the logic based on task type there.